src/include/server/mir/scene/idle_hub.h

```diff
@@ -68,6 +68,10 @@ class IdleHub
 
     virtual void unregister_interest(IdleStateObserver const& observer) = 0;
 
+    virtual void inhibit_idle() = 0;
+
+    virtual void resume_idle() = 0;
```

Comment:
This is a good suggestion. I will be taking this approach instead. I hadn't considered the issues that would occur with two surfaces calling for `resume_idle()`. While I think the way my implementation works would be fine in the common case (if a surface loses focus, call `resume_idle()`, but if another surface comes into focus and wants to inhibit idle, that calls `inhibit_idle()`, thus retaining the state, this could all easily fall apart upon an exception.